File: 0016-最接近的三个整数之和/Solution.java
import java.util.Arrays;

class Solution {
    public int threeSumClosest(int[] nums, int target) {
        Arrays.sort(nums);
        int res = nums[0] + nums[1] + nums[2];
        for (int i = 0; i < nums.length-2; i++) {
            int left=i+1, right=nums.length-1;
            while(left < right){
                int sum = nums[i] + nums[left] + nums[right];
                if(sum == target){
                    return sum;
                }else if(sum < target){
                    left++;
                }else{
                    right--;
                }
                res = Math.abs(sum-target)<Math.abs(res-target)? sum: res;
            }
        }
        return res;
    }
}

File: 0016-最接近的三个整数之和/solution.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        res = nums[0] + nums[1] + nums[2]
        for i in range(len(nums)):
            l, r = i+1, len(nums)-1
            while l < r:
                tmp = nums[i] + nums[l] + nums[r]
                if abs(target-res) > abs(target-tmp):
                    res = tmp
                if tmp > target:
                    r -= 1
                elif tmp < target:
                    l += 1
                else:
                    return tmp
        return res
